cloudtools/aws/vpc.py

```diff
@@ -25,16 +28,32 @@ def ip_available(region, ip):
         return True
 
 
-def get_avail_subnet(region, subnet_ids, availability_zone=None):
+@lru_cache(100)
+def get_all_subnets(region, subnet_ids):
     vpc = get_vpc(region)
-    subnets = vpc.get_all_subnets(subnet_ids=subnet_ids)
-    subnets = [s for s in subnets if s.available_ip_address_count > 0]
-    if availability_zone:  # pragma: no branch
-        subnets = [s for s in subnets if s.availability_zone ==
-                   availability_zone]
-    subnets.sort(key=lambda s: s.available_ip_address_count)
-    if not subnets:
+    return vpc.get_all_subnets(subnet_ids=subnet_ids)
+
+
+def get_avail_subnet(region, subnet_ids, availability_zone):
+    subnets = [s for s in get_all_subnets(region, tuple(subnet_ids))
+               if s.available_ip_address_count > 0 and
+               s.availability_zone == availability_zone]
+    pending_spot_req = [sr for sr in get_active_spot_requests(region) if
+                        sr.state == 'open']
+    usable_subnets = []
+    UsableSubnet = namedtuple("UsableSubnet", ["subnet", "usable_ips"])
+    for s in subnets:
+        # Subtract pending requests from available IP count
+        pending_req = [sr for sr in pending_spot_req if
+                       sr.launch_specification.subnet_id == s.subnet_id]
+        usable_ips = s.available_ip_address_count - len(pending_req)
+        if usable_ips > 0:
```

Comment:
should we check that this is > 1 or 2 instead of 0, to leave us some small margin of error?